### PR TITLE
 Remove setHeatCoolIfNeeded()

### PR DESCRIPTION
**Pull request by leehendricks**
https://github.com/chrisjshull/homebridge-nest/issues/47

### CODE REVIEW

File: lib/nest-thermostat-accessory.js [hunk 1]
```diff
@@ -383,21 +383,15 @@ NestThermostatAccessory.prototype.setTargetTemperatureDebounced = debounce(funct
...
-    ? Promise.resolve()
-    : this.updateDevicePropertyAsync("hvac_mode", 'heat-cool', "target heating cooling");
-};
-
 NestThermostatAccessory.prototype.setCoolingThresholdTemperatureDebounced = debounce(function (targetTemperature, callback) {
-  return this.setHeatCoolIfNeeded()
-    .then(() => this.updateTargetWithCorrectUnitsAsync('target_temperature_high_', targetTemperature, "cooling threshold temperature"))
+  return Promise.resolve()
+    .then(() => this.updateTargetWithCorrectUnitsAsync('target_temperature_', targetTemperature, "cooling threshold temperature"))
     .asCallback(callback);
```
Comment (chrisjshull) [inline lib/nest-thermostat-accessory.js]: this simplifies to 
```js
return this.updateTargetWithCorrectUnitsAsync('target_temperature_', targetTemperature, "cooling threshold temperature").asCallback(callback);
``` 
(ditto below)

File: lib/nest-thermostat-accessory.js [hunk 2]
```diff
@@ -383,21 +383,15 @@ NestThermostatAccessory.prototype.setTargetTemperatureDebounced = debounce(funct
...
-  return this.device.hvac_mode === 'heat-cool'
-    ? Promise.resolve()
-    : this.updateDevicePropertyAsync("hvac_mode", 'heat-cool', "target heating cooling");
-};
-
 NestThermostatAccessory.prototype.setCoolingThresholdTemperatureDebounced = debounce(function (targetTemperature, callback) {
-  return this.setHeatCoolIfNeeded()
-    .then(() => this.updateTargetWithCorrectUnitsAsync('target_temperature_high_', targetTemperature, "cooling threshold temperature"))
+  return Promise.resolve()
+    .then(() => this.updateTargetWithCorrectUnitsAsync('target_temperature_', targetTemperature, "cooling threshold temperature"))
```
Comment (chrisjshull) [inline lib/nest-thermostat-accessory.js]: I think this needs to be target_temperature_high_ for those who *do* use auto mode. (ditto below with target_temperature_low_)

File: lib/nest-thermostat-accessory.js (hunk 1 above)
Comment (leehendricks) [reply lib/nest-thermostat-accessory.js]: duh. thanks!

File: lib/nest-thermostat-accessory.js (hunk 2 above)
Comment (leehendricks) [reply lib/nest-thermostat-accessory.js]: yes, i just discovered that. revising.

**Comment by leehendricks** [comment]
All fixed. Let me know if I missed anything else. First time in this repo - your code is very clean. 👍

**Comment by chrisjshull** [comment]
Thanks! (credit where due; most of this code was inherited) 
I'm aiming to test and deploy in the next day or so.

**Comment by vmiyoshi** [comment]
I fixed the issue in my home using @leehendricks "nest-thermostat-accessory.js" file. 
However another issue I hadn't seen anyone mention was beside siri changing from "COOL" to AUTO," the assistant would also always give you the "AUTO" settings if you ask her what the thermostat was set to. 
- Examples - 
"hey siri set temperature to" or "hey siri set thermostat to" works now with @leehendricks "nest-thermostat-accessory.js" file in place. 
however 
"hey siri what is my thermostat set to" always shows what "AUTO" is set to (which is the range), only unless the thermostat is actively cooling.

**Comment by leehendricks** [comment]
@vmiyoshi Did my fix introduce the bug you mentioned or is that a separate issue?

File: lib/nest-thermostat-accessory.js [hunk 3]
```diff
@@ -383,22 +383,14 @@ NestThermostatAccessory.prototype.setTargetTemperatureDebounced = debounce(funct
...
 NestThermostatAccessory.prototype.setCoolingThresholdTemperatureDebounced = debounce(function (targetTemperature, callback) {
-  return this.setHeatCoolIfNeeded()
-    .then(() => this.updateTargetWithCorrectUnitsAsync('target_temperature_high_', targetTemperature, "cooling threshold temperature"))
-    .asCallback(callback);
+  const setting = this.device.hvac_mode === 'heat-cool' ? 'target_temperature_high_' : 'target_temperature_';
+  return this.updateTargetWithCorrectUnitsAsync(setting, targetTemperature, "cooling threshold temperature").asCallback(callback);
 }, 5000);
 
 NestThermostatAccessory.prototype.setHeatingThresholdTemperatureDebounced = debounce(function (targetTemperature, callback) {
-  return this.setHeatCoolIfNeeded()
```
[the remark this answers is not in the document]
Comment (leehendricks) [reply lib/nest-thermostat-accessory.js]: Thanks @chrisjshull. Appreciate the quick turnaround.

**Comment by drewcovi** [comment]
I believe this introduced a new issue, wherein I’m no longer able to set temperature ranges in the Home app. They’re simply ignored... I’ll take a stab at a PR in the next few weeks, but I’d appreciate it if someone could quickly set a temperature range (autochangeover) in the home app and verify that it sticks. 😄

**Pull request merged**